### ISSUE TITLE
what is action_loss

### ISSUE THREAD

**Issue by christofjan** [bryanyzhu/Hidden-Two-Stream#25]
Hi Sir,
            During training i encountered the word action_loss, what is it and how it is evaluated? and I would like to plot graph for train loss vs iterations. I am not able to plot graphs using default scripts. so, could you please tell me that what parameter should i consider for the train_loss in the log file.

thanks..

**Comment by bryanyzhu** [comment]
Hi, sorry for the late reply. The action_loss is defined in [here](https://github.com/bryanyzhu/Hidden-Two-Stream/blob/master/models/ucf101_split1_unsup_end/end_train_val.prototxt#L8166-L8187). It is the action classification loss, which is a softmax loss. It is evaluated as accuracy. For example, you have 10 videos, and predict 6 correct, thus the accuracy is 60%.

If you want to plot a graph to visualize, you can save the log into a file, something like:

`python main.py > log.txt &`

And then read the `log.txt` file using python, extract the accuracy, and plot it. In that case, you can customize it and plot anything you like.

I paste my code below, hope it is helpful. It should be straightforward to read.

```
import os,sys
import numpy as np
import matplotlib.pyplot as pp

log_file = "log.txt"
f_log = open(log_file, "r")
log_lines = f_log.readlines()
print len(log_lines)

test_loss_all = []
for line_num in xrange(len(log_lines)):
    line = log_lines[line_num]
    if "Test net output #4:" in line:
	line_info = line.split()
	test_loss = line_info[-1]
	if test_loss != "evict":
       	    test_loss_all.append(test_loss)

losses = np.asarray(test_loss_all)
pp.plot(losses, 'x')
pp.show()
```

**Comment by christofjan** [comment]
Hi, Thank you so much for the code.. You helped me a lot in my research work. If you don't mind, I am going to write your name in my research paper acknowledgement.. Thanks a lot..

**Comment by bryanyzhu** [comment]
No, I don't mind at all. That would be an honor. I am glad I could help.
